buffer/src/test/java/io/netty/buffer/PoolArenaTest.java

```diff
@@ -107,4 +110,17 @@ public final void testAllocationCounter() {
         Assert.assertEquals(1, metric.numNormalDeallocations());
         Assert.assertEquals(1, metric.numNormalAllocations());
     }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public final void testDirectArenaMemoryCopy() {
+        PooledByteBuf<ByteBuffer> src =
+                (PooledByteBuf<ByteBuffer>) PooledByteBufAllocator.DEFAULT.directBuffer(512);
+        PooledByteBuf<ByteBuffer> dst =
+                (PooledByteBuf<ByteBuffer>) PooledByteBufAllocator.DEFAULT.directBuffer(512);
+        // This causes the internal reused ByteBuffer duplicate limit to be set to 128
+        dst.writeBytes(ByteBuffer.allocate(128));
+        // Ensure internal ByteBuffer duplicate limit is properly reset (used in memoryCopy non-Unsafe case)
+        dst.chunk.arena.memoryCopy(src.memory, 0, dst, 512);
```

Comment:
@njhill we need to call `src.release()` and `dst.release()` as well.

Reply:
@normanmaurer done